typegate/src/utils.ts

```diff
@@ -214,3 +225,82 @@ export function collectFieldNames(tg: TypeGraph, typeIdx: number) {
   }
   return { title: typ?.title, fields: [] };
 }
+
+/**
+ * base64 decode and untar at cwd/{dir}
+ */
+export async function uncompress(dir: string, tarb64: string) {
```

Comment:
suggestion, as you compress in rust, maybe it is interesting to uncompress in rust so we are sure this is symmetric?

Reply:
Wouldn't that mean that we have to expose the lib function to deno ? I cannot see any other way.

Reply:
@afmika yes, you could create a new crates in libs and use it both in meta-cli and typegate native